build: improve overview page (#15819)
docker/docs: build/index.md

### Packaging your software

> Build and package your application to run it anywhere, from your local Docker
> Desktop, to Docker Engine and Kubernetes on the cloud:

Reviewer: ```suggestion
Build and package your application to run it anywhere: locally using Docker
Desktop, or in the cloud using Docker Engine and Kubernetes.
```

[Packaging your software](building/packaging.md){: .button .outline-btn }

### Choosing a build driver

Run Buildx with different configurations depending on the scenario you are
working on, regardless of whether you are using your local machine or a remote
cluster, all from the comfort of your local working environment:

[Choosing a build driver](building/drivers/index.md){: .button .outline-btn }

### Optimizing builds with cache

Improve build performance by using a persistent shared build cache to avoid

> repeating costly operations such as package installations, downloading files,
> or code build steps:

Reviewer: ```suggestion
repeating costly operations such as package installs, file downloads,
or code build steps:
```
